src/components/Droplist/Droplist.tsx

```diff
@@ -0,0 +1,99 @@
+import classNames from 'classnames';
+import { FC, useRef, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { getSearchWith } from '../../helpers/searchHelper';
+import arrowDown from '../../assets/svg/arrowDown.svg';
+
+import './droplist.scss';
+
+type Props = {
+  options: string[];
+  startValue: string;
+  label: string;
+  searchParamsKey: string;
+};
+
+export const Droplist: FC<Props> = ({
+  options,
+  startValue,
+  label,
+  searchParamsKey,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggle = () => setIsOpen(!isOpen);
+  const [searchParams] = useSearchParams();
+  const [value, setValue] = useState(startValue);
+  const bntRef = useRef<HTMLButtonElement>(null);
+
+  document.addEventListener('click', (e) => {
+    if (e.target !== bntRef.current) {
+      setIsOpen(false);
+    }
+  });
+
+  document.addEventListener('keyup', (e) => {
+    if (e.key === 'Tab' || e.key === 'Escape') {
+      setIsOpen(false);
+    }
+  });
```

Comment:
Add listeners inside useEffect and remove them on component unmount

```
useEffect(() => {
  // add listeners
  return () => {
    // remove listeners
  }
}, [])
```

Reply:
done